python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -120,6 +120,22 @@ def __init__(self, provider_config, cluster_name):
         # excessive DescribeInstances requests.
         self.cached_nodes = {}
 
+    def _gc_tag_cache(self):
```

Comment:
This seems complicated. Why not instead make all accesses to the tag cache fall back on looking up the tags in case of failure?

Reply:
You mean making an API call to get the tags?

Reply:
Yes, or returning empty dict if the node is terminated.

Hmm another option is to never GC the tags; it seems harmless enough to keep them forever, since the number of nodes will never be that large.

Reply:
An empty dict would be an inconsistent state so I'd prefer to avoid that. I guess just leaking the tag cached isn't the end of the world...